### PR TITLE
Upgrade to programmatic pipeline and the new error handler

### PR DESCRIPTION
**Pull request by xtreamwayz**
This PR adds support for zendframework/zend-expressive#396.

- [x] Convert to the programmatic pipeline
- [x] Enable new error handler
- [x] Update header license
- [x] Update link to docs
- [x] Remove routes from config/routes.php for minimal install
- [x] Remove PHP 5.5 support
- [x] Update to PHPUnit 5.6
- [x] Update tests
- [x] Enable Whoops 2 error handler

TODO after zend-expressive 1.1 is released:
- [ ] Remove 1.1.x-dev branch from zend-expressive version in composer.json

### CODE REVIEW

**Comment by xtreamwayz** [comment]
@weierophinney I've started with updating the skeleton to zend-expressive 1.3 and programmatic configuration, It seems to work.

File: test/ExpressiveInstallerTest/RoutersTest.php [hunk 1]
```diff
@@ -78,7 +79,7 @@ public function testRouter(
             $expectedRouter,
             $config['dependencies']['invokables'][Router\RouterInterface::class]
         );
-        $this->assertEquals($expectedRoutes, $config['routes']);
+        //$this->assertEquals($expectedRoutes, $config['routes']);
```
Comment (weierophinney) [inline test/ExpressiveInstallerTest/RoutersTest.php]: Why is this commented out, instead of removed outright?

File: .travis.yml [hunk 2]
```diff
@@ -17,26 +17,13 @@ notifications:
...
   include:
-    - php: 5.5
-      env:
-        - DEPS=lowest
-    - php: 5.5
-      env:
-        - DEPS=latest
-    - php: 5.6
-      env:
-        - DEPS=lowest
```
Comment (xtreamwayz) [inline .travis.yml]: Also do we need hhvm builds? They fail anyway.

File: README.md [hunk 3]
```diff
@@ -79,7 +79,7 @@ scripts** with `--no-scripts`, otherwise it will remove the installer and all
 tests.
 
 ```bash
-$ composer install --no-scripts
+$ composer update --no-scripts
```
Comment (weierophinney) [inline README.md]: :+1:

**Comment by weierophinney** [comment]
Thanks, @xtreamwayz\!

**Pull request merged**